### PR TITLE
Add Ipopt L1 solver factory

### PR DESCRIPTION
**Pull request by eslickj**
## Summary/Motivation:

This adds a solver factory for ipopt_l1 and fills in some config blocks to make it easier to configure with the IDAES config system.  **To use this,** you need to ``from idaes.core.solvers import ipopt_l1`` 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/config.py [hunk 1]
```diff
@@ -249,6 +249,44 @@ def _new_idaes_config_block():
         ),
     )
 
+    cfg.declare(
+        "ipopt_l1",
+        pyomo.common.config.ConfigBlock(
+            implicit=False,
+            description="Default config for 'ipopt' solver",
```
Comment (jsiirola) [inline idaes/config.py:256]: Should all these descriptions reference `ipopt_l1` instead of `ipopt`?

**Review by jsiirola — COMMENTED** [review]
One question on documentation, but otherwise looks good.

File: idaes/config.py (hunk 1 above)
Comment (eslickj) [reply idaes/config.py:256]: Yeah.  I'll fix it.

**Review by jsiirola — APPROVED** [review]

**Review by andrewlee94 — APPROVED** [review]
Mostly a rubber stamp.

**Review by viiibhav — APPROVED** [review]
Looks good

**Pull request merged**